include/swift/AST/DiagnosticsSema.def

```diff
@@ -992,6 +992,9 @@ ERROR(missing_address_of,none,
 ERROR(missing_address_of_yield,none,
       "yielding mutable value of type %0 requires explicit '&'",
       (Type))
+ERROR(missing_property_wrapper_unwrap,none,
+      "wrapped value of type %0 must be unwraped to value of type %1",
```

Comment:
Terminology here: are we going to consider the wrapped value as "wrapped" when it's used without the `$`, and then "unwrapped" when it's used with the `$`?

Semantically, adding the $ stops the property from being unwrapped.

Also small typo: `unwraped` -> `unwrapped`

Reply:
Good point, I did get this backwards. That said, I can’t think of a good way to talk about this the other way around. Saying something like: “the property wrapper must not be unwrapped” sounds very strange because all property wrappers are implicitly unwrapped so the user didn’t do anything to unwrap it. It gets even weirder for people new to the feature when the suggested fix is to add a $ to *prevent* the implicit unwrapping. Any idea for what the diagnostic for this should be?

Reply:
Thanks Harlan! I actually haven't looked at the message... It looks like it should be `extraneous_property_unwrap` and message should get adjusted accordingly e.g.  `extraneous property %0 unwrap to value of type %1, use `$` to refer to wrapper type.` where `$0` is `DeclName` and $1 is `Type`. @harlanhaskins WDYT?

Reply:
or `cannot convert type %0 to expected property wrapper type %1, use $ to refer to wrapper type.`

Reply:
Hm, I know we have precedent for `extraneous` but I think we can simplify this text. Maybe `property %0 will be unwrapped to value of type %1, use '$' to refer to the property's wrapper`

Reply:
Or maybe `use '$' to refer to the @%2 property wrapper`, where `%2` is the `Type` of the wrapper?

Reply:
+1